### PR TITLE
Fix MinGW Travis build issues

### PR DESCRIPTION
**Pull request by neptoess**
Relative discussion in #3130

### CODE REVIEW

**Comment by edenhill** [comment]
Perfect, thank you!

File: packaging/mingw-w64/configure-build-msys2-mingw-static.sh [hunk 1]
```diff
@@ -50,6 +50,3 @@ cp ./librdkafkacpp-static.a ../dest/lib/librdkafka++-static.a
 popd
 rm -rf ./mergescratch
 
-export PATH="$PWD/dest/bin:/mingw64/bin/:${PATH}"
```
Comment (edenhill) [inline packaging/mingw-w64/configure-build-msys2-mingw-static.sh:53]: late comment, but these should be moved for the dynamic build script too.

**Comment by edenhill** [comment]
Hey @neptoess , pinging you here in lack of better communication alternatives:
The dynamic mingw build is failing (silently) because msys2 now uses gcc 11 which introduces a new `_ZSt28__throw_bad_array_new_lengthv` symbol,  but the libstdc++ on the system is compiled with an older gcc that  does not have this symbol, so trying to execute the test runner fails silently.
It is basically this issue:
https://github.com/stan-dev/pystan/issues/294

But I'm not sure that solution (symlink/replace) is viable for our msys2 builder.

Do you have any ideas?

https://app.travis-ci.com/github/edenhill/librdkafka/jobs/564513967

**Comment by neptoess** [comment]
[v1.9.0-PRE11.build.log](https://github.com/edenhill/librdkafka/files/8363343/v1.9.0-PRE11.build.log)
Upgraded my msys2 packages and did the build. As the old joke goes, "works on my machine".

I also just realized that the build you linked was the dynamic build (pinging @ed-alertedh @edbordin since he was involved there), but the static build worked (https://app.travis-ci.com/github/edenhill/librdkafka/jobs/564513968). The build log you linked seems to cut off without even attempting to run the tests. Bizarre.

Adding a `$msys2 pacman -Q` to the end of `travis-before-install.sh` might be useful, so we can easily see exactly which packages, and which versions, are installed during the build.

**Pull request merged**